src/platform/plugins/shared/visualizations/public/actions/register_actions.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import { CONTEXT_MENU_TRIGGER } from '@kbn/embeddable-plugin/public';
+import { ADD_PANEL_TRIGGER, type UiActionsStart } from '@kbn/ui-actions-plugin/public';
+import { ACTION_EDIT_IN_LENS, ADD_AGG_VIS_ACTION_ID } from './constants';
+import { TypesStart } from '../vis_types/types_service';
+
+export function registerActions(
+  uiActions: UiActionsStart,
+  data: DataPublicPluginStart,
+  types: TypesStart
+) {
+  uiActions.addTriggerActionAsync(CONTEXT_MENU_TRIGGER, ACTION_EDIT_IN_LENS, async () => {
+    const { EditInLensAction } = await import('./actions_module');
+    return new EditInLensAction(data.query.timefilter.timefilter);
+  });
+
+  uiActions.addTriggerActionAsync(ADD_PANEL_TRIGGER, ADD_AGG_VIS_ACTION_ID, async () => {
+    const { AddAggVisualizationPanelAction } = await import('./actions_module');
+    return new AddAggVisualizationPanelAction(types);
+  });
```

Comment:
Instead of importing the `actions_modules` isn't better to directly import the Action directly from its file?
Since we are async loading these to reduce a bit the initial loading we can probably also gain a bit during the action loading if we async load the Actions independently, WDYT?

Reply:
that makes sense since the actions are for different triggers. Resolved with 4bcd8f085f83b14b4d6131d311dbc6ed413641b3